shared/src/conversions.rs

```diff
@@ -1,12 +1,20 @@
-use num::bigint::Sign;
+use anyhow::{anyhow, Result};
 use num::BigInt;
+use num::{bigint::Sign, Zero};
 use num::{BigRational, ToPrimitive as _};
 use primitive_types::U256;
 
 pub fn big_rational_to_float(ratio: &BigRational) -> Option<f64> {
     Some(ratio.numer().to_f64()? / ratio.denom().to_f64()?)
 }
 
+pub fn big_rational_to_u256(ratio: &BigRational) -> Result<U256> {
+    if ratio.denom() == &BigInt::zero() {
+        return Err(anyhow!(" Division by 0 in BigRational to U256 conversion"));
+    }
```

Comment:
Just wondering if that is the canonical representation of zero in BigRational or if for example numer being 0 with nonzero denom could also be it. Could maybe use https://docs.rs/num-traits/0.2.12/num_traits/identities/trait.Zero.html ?
Or even better use BigRational::to_integer and then check if that is zero or conver it.

Reply:
I recall there is a `num::zero()` as well.